### DISCUSSION TITLE
Create  a Helper Tool (maybe in a CommunitToolkit Gallery?) for choosing the best framework (Maui/Winui/WPF...) for a new application 😍

### DISCUSSION THREAD

**Discussion by DevTKSS** [CommunityToolkit/dotnet#1005] [Ideas]
Maybe not exactly right repo, so please feel free to advise me a better place for this idea!

I am coming from WinForms (framework and vba, so its time for learning actual things now 👍 ) and vb.net and went to the learning pathes on Microsoft learn. There was a kind of guide for which framework to use, but to me it seems to be a bit outdated (recommending WinForms?!?)

The Tool I am thinking of would maybe be found in microsoft learn or also in a community toolkit gallery or similar, where new developers might be search by default.
Criteria it should ask:
- Dependencys/Requirement to connect/stay with .net versions (e.g. Framework?)
- -> take under consideration, that this dependency can not be moved to .net, what would the dev should be recommended beside quit its current job ;)
- Target Framework/Device
- -> Windows (include Verision, which might be also a criteria for choosing)
- -> Skia
- -> IOS
- -> Android
- -> WASM (I do not mean blazor with that)
- ->Blazor/Razor acceptable
- ->Linux
- ->Mac

- Language for UI
- -> Some of the new UI Frameworks are enabling using c# for them, which would be a criteria to only have one to learn, not 2 or more

- Needed Connectivity like Rest-API or Databases could be a criteria for the dev, but the resources for this are kind of spread everywhere and not as understandable as it could be, so maybe more broken down and simlyfied?

- Extending the "see the choosen UI Framework live" capability like in the gallery apps, just that the user could have this comparement side by side or collected in one app.

making this tool a open source thing would enable other providers to add themself to this for improvement, but they should use .net of course.
In the actual guide I am missing ones like uno which is mentioned in the CommunityToolkit/Windows repo for winui which are providing the capability for more plattforms/devices than a Maui one for example.
The tool could also include a better overview of extension package providers, like telerik or synfusion are mentioned for maui, but there should be added the information if using this is free or paid, because maybe the dev that is using this tool for decision is just starting and not able to just spend thousands for using nice controls (referring to Telerik for example)?

additionally maybe there could also be links to the many different gallery applications for the then choosed ui (Winui/WPF/UWP...) like the MauiIsland galley in microsoft store.

More feature ideas could be added naturally!

**Comment by Sergio0694** [comment]
This feels more like it should be a docs page, which already exists: https://learn.microsoft.com/en-us/windows/apps/get-started/.

**Reply by DevTKSS** [reply]
Was thinking about something interactive...
For example ckecking some relevant criteria and get a filtered list...
Like, all the great things that was introduced in the .net conf from the 2.day on, if you are lucky, maybe you will find them somewhere hidden in documentation, but not on such a entry page.
Enchancement would Maybe also be, to not have to read ALL of the content of ALL options, but only the ones that are relevant for the criteria that was provided

The docs page you mentioned is kind of static in my opinion and the options it is showing could also be improved maybe.
